app/assets/src/components/views/SampleViewV2/ReportTable.jsx

```diff
@@ -364,10 +401,38 @@ class ReportTable extends React.Component {
   };
 
   toggleExpandGenus = ({ taxId }) => {
-    const { expandedGenusIds } = this.state;
+    const { data } = this.props;
+    const { expandedGenusIds, expandAllOpened } = this.state;
     expandedGenusIds.delete(taxId) || expandedGenusIds.add(taxId);
 
-    this.setState({ expandedGenusIds: new Set(expandedGenusIds) });
+    let newExpandedAllOpened = expandAllOpened;
+    if (expandedGenusIds.size === data.length) {
```

Comment:
Thanks! The logic makes more sense now.

Nit:
I notice that the value of expandAllOpened depends not only on how many genuses are opened, but also on the previous value of expandAllOpened.

If there are 10 rows, and 5 are opened, and I click on the expandAll button, the behavior (whether all rows open or close) depends on whether I previously clicked the expandAll button. 

I feel like the behavior should just depend on the number of rows opened. So if there are 5/10 rows opened, clicking expandAll should always open all rows. It should only close all rows if 10/10 rows are opened.

I'll leave it up to you, but not knowing whether the button will expand all or collapse all feels more confusing to me.

Reply:
It feels more intuitive to me that if the users click in the expand/collapse all, that the next time they click it should have the opposite behavior (unless that behavior makes no sense because the rows are all expanded or all closed).
We can leave that as UX topic for @happyimadesignr to chime on once she does here design review.